Added downscaling and option to remove loading animation to SetImage

Taffeeko opened the pull request:
Added parameter for if the image should be downscaled along with an parameter to remove the loading animation.

Shouldn't break anything using SetImage as an overload method has been introduced that invokes SetImage with it's original behaviours

I've made this PR to help with Nya 0.4.0 along with a fork of the DrinkWater mod I'm planning to release

Taffeeko committed:
Added compression to SetImage

Auros commented on BeatSaberMarkupLanguage/BeatSaberUI.cs:
```diff
@@ -341,13 +349,29 @@ public static void SetImage(this Image image, string location)
                         GameObject.DestroyImmediate(stateUpdater);
-                    image.sprite = Utilities.LoadSpriteRaw(data);
+                    
+                    var originalImage = System.Drawing.Image.FromStream(new MemoryStream(data));
```
This memory stream is never disposed!

Taffeeko replied:
Whoopsie, thought it would do that by itself

Taffeeko committed:
Properly dispose of memory stream

I think

Taffeeko committed:
Added overload method

Okay okay, before I did my PR I swear I did test my changes.
I just did an incredibly poor job at testing.

Taffeeko committed:
Changed wording from compress to downscale image

Downscaling counts as a method of compression, right?
To try and hide that I don't know the answer to this question, I've changed the wording from compress to downscale. It's more specific anyways

Taffeeko committed:
Moved scaling off main thread and added more scaling options

Taffeeko committed:
Fixed bad scaled image if larger than original image

By means beyond me the scaled image can sometimes be larger, file size wise, than the original image. Because I thought I was being super cool and resourceful by reusing the memory stream the larger image wouldn't be able to fit in it resulting in god knows what being returned by the downscale method
